config/src/test/java/org/axonframework/config/DefaultConfigurerLifecycleOperationsTest.java

```diff
@@ -65,6 +65,30 @@ void testStartLifecycleHandlersAreInvokedInAscendingPhaseOrder() {
         lifecycleOrder.verify(phaseOverNineThousandHandler).start();
     }
 
+    @Test
+    void testStartLifecycleHandlerConfiguredThroughConfigurerAreInvokedInAscendingPhaseOrder() {
+        Configurer testSubject = DefaultConfigurer.defaultConfiguration();
+
+        LifecycleManagedInstance phaseZeroHandler = spy(new LifecycleManagedInstance());
+        LifecycleManagedInstance phaseOneHandler = spy(new LifecycleManagedInstance());
+        LifecycleManagedInstance phaseTenHandler = spy(new LifecycleManagedInstance());
+        LifecycleManagedInstance phaseOverNineThousandHandler = spy(new LifecycleManagedInstance());
+
+        testSubject.onStart(0, phaseZeroHandler::start);
```

Comment:
Nit: In these tests we expect the LifeCycleHanlders to be handled in the order they are set here. If there would ever be a bug that causes the phase to be ignored and these handlers to simply be invoked in the order they are defined, then these tests would not catch that. As such, although it's quite a remote chance something like this would happen, it might be beneficial to change the order of the next 4 lines somewhat.

Reply:
Good thinking! I've reordered them just now.